src/vtl.ts

```diff
@@ -188,6 +188,14 @@ export class VTL {
       case "BinaryExpr":
         // VTL fails to evaluate binary expressions inside an object put e.g. $obj.put('x', 1 + 1)
         // a workaround is to use a temp variable.
+        if (node.op === "in") {
+          // !! right[left]
+          return this.var(
+            `!!${this.eval(node.right)}[${this.eval(node.left)}]`
```

Comment:
VTL has the `!!` operator?

Reply:
not not

Though now I am concerned that this would not work with a falsely value...

Any idea how to prove existence?

Reply:
Use HashMap .has

Reply:
`has` returns false.

Reply:
`.containsKey`

https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#containsKey-java.lang.Object-

and `arr.length >= [left]` for array.

Reply:
thats how I am recreating 

`index in array` 
`1 in [1,2] => true`
`2 in [1,2] => false`

in typescript

Reply:
I don't understand.

Reply:
`In` for array checks for the existence of indices.